ansible/library/port_alias.py

```diff
@@ -101,33 +113,67 @@ def get_portmap(self):
                             alias_index = index
                         if 'speed' in text:
                             speed_index = index
+                        if 'role' in text:
+                            role_index = index
             else:
                 if re.match('^Ethernet', line):
                     mapping = line.split()
                     name = mapping[0]
+                    if (role_index != -1) and (len(mapping) > role_index):
+                        role = mapping[role_index]
+                    else:
+                        role = 'Ext'
                     if alias_index != -1 and len(mapping) > alias_index:
                         alias = mapping[alias_index]
                     else:
                         alias = name
-                    aliases.append(alias)
-                    portmap[name] = alias
-                    aliasmap[alias] = name
-                    if (speed_index != -1) and (len(mapping) > speed_index):
-                        portspeed[alias] = mapping[speed_index]
+                    if role == 'Ext':
+                        aliases.append(alias)
+                        portmap[name] = alias
+                        aliasmap[alias] = name
+                        if (speed_index != -1) and (len(mapping) > speed_index):
+                            portspeed[alias] = mapping[speed_index]
 
         return (aliases, portmap, aliasmap, portspeed)
 
 def main():
     module = AnsibleModule(
         argument_spec=dict(
-            hwsku=dict(required=True, type='str')
+            hwsku=dict(required=True, type='str'),
+            num_asic=dict(type='int', required=False)
         ),
         supports_check_mode=True
     )
     m_args = module.params
     try:
+        aliases = []
+        portmap = {}
+        aliasmap = {}
+        portspeed = {}
         allmap = SonicPortAliasMap(m_args['hwsku'])
-        (aliases, portmap, aliasmap, portspeed) = allmap.get_portmap()
+        # When this script is invoked on sonic-mgmt docker, num_asic 
+        # parameter is passed.
+        if m_args['num_asic'] is not None:
+            num_asic = m_args['num_asic']
+        else:
+            # When this script is run on the device, num_asic parameter
+            # is not passed.
+            try: 
+                num_asic = multi_asic.get_num_asics()
+            except Exception, e:
+                num_asic = 1
+        for asic_id in range(num_asic):
+            if num_asic == 1:
```

Comment:
if we are trying to set the asic_id as None for handling single asic case, should this be num_asic == 0 ?

Reply:
In ansible script, where we would want to pass num_asic parameter, we will either use the value defined for a DUT from veos_vtb or default to the value 1. We could have for loops based on num_asic hence using it as 1 by defualt.

Reply:
I have made a fix to avoid updating asic_id to make it easier to read as by default, asic_id will be None.

Reply:
True right, got mixed up with the variables num_asic and asic_id !

Reply:
My previous commit was wrong, we will need the asic_id to be set to None or update the code:
if num_asic == 1:
    (aliases_asic, portmap_asic, aliasmap_asic, portspeed_asic) = allmap.get_portmap()
 else: 
 
 To avoid this if-else check, i have kept the asic_id updating to None as is.